cibyl/plugins/openstack/sources/elasticsearch.py

```diff
@@ -0,0 +1,198 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+
+import logging
+import re
+
+from cibyl.models.attribute import AttributeDictValue
+from cibyl.models.ci.base.job import Job
+from cibyl.plugins.openstack.deployment import Deployment
+from cibyl.sources.source import speed_index
+from cibyl.utils.filtering import IP_PATTERN
+
+LOG = logging.getLogger(__name__)
+
+
+class ElasticSearch:
```

Comment:
This is one of those things that I would notify somewhere. "If you are implementing a source, separate the CI and OSP functions into two different classes."

If I had not seen this pull request, I would have committed the same mistake with Zuul later on.

Reply:
Good point